Source/CarthageKit/Project.swift

```diff
@@ -540,38 +540,23 @@ public final class Project {
 					.map { (cartfile: Cartfile) -> DependencyGraph in
 						return [ dependency.project: Set(cartfile.dependencies.map { $0.project }) ]
 					}
+					.concat(SignalProducer(value: [ dependency.project: Set() ]))
+					.take(1)
 			}
-			.collect()
-			.flatMap(.Latest) { (graphs: [DependencyGraph]) -> SignalProducer<Dependency<PinnedVersion>, CarthageError> in
-				let graph = graphs.reduce([:]) { (graph: DependencyGraph, item: DependencyGraph) -> DependencyGraph in
-					var result = graph
-					for (dependency, projects) in item {
-						var recursive = projects
-						for project in projects {
-							if let projects = result[project] {
-								recursive.unionInPlace(projects)
-							}
-						}
-						
-						result.updateValue(recursive, forKey: dependency)
-						
-						for (project, projects) in result {
-							if projects.contains(dependency) {
-								result.updateValue(projects.union(recursive), forKey: project)
-							}
-						}
-					}
-					return result
+			.reduce([:]) { (working: DependencyGraph, next: DependencyGraph) in
```

Comment:
@mdiep I couldn't figure out how to use `mergeGraphs` here—it would seem that there's two types of `DependencyGraph` in CarthageKit.